lib/bolt/cli.rb

```diff
@@ -559,6 +563,40 @@ def initialize_project
       ok ? 0 : 1
     end
 
+    def migrate_project
+      if inventory.version == 2
+        ok = true
+      else
+        inventoryfile = config.inventoryfile || config.boltdir.inventory_file
```

Comment:
While i dont think you necessarily need to worry about inventory from BOLT_INVENTORYenvironment variable, I recently found it slightly more complicated that you might initially think to find the actual source of inventory data... https://github.com/puppetlabs/bolt/blob/ab4a96649e8e6a74393527ea8b3b5a9be40a83f8/lib/bolt/cli.rb#L269-L282 maybe we should share that logic?

Reply:
It seems like the inventory is sourced the same way, this just has some extra validation thrown in with the file stat? If we want this to be reusable would it make sense to have it in `config`?

Reply:
For some reason i think I remember finding a case where the config.boldir.inventoryfile is different from the array in config.default_inventoryfile. Also you will for sure want to verify that the inventory file exists. If you do `bolt project migrate` currently with a non-existent inventoryfile you get a ENOENT stacktrace.

Reply:
Got it. Would you recommend I pull this functionality out into another method or something else?

Reply:
I think a new method on the CLI method seems reasonable, this is the second time we want to know *where* inventory will be loaded from without loading it. I based the algo I posted on https://github.com/puppetlabs/bolt/blob/b12168d670853030657156b7242585e5bf5cd5fa/lib/bolt/inventory.rb#L46-L61

Reply:
I'm also really curious what that case would be since it seems like `config.default_inventoryfile` shouldn't be different from `config.boltdir.inventory_file` since the former just returns `[@boltdir.inventory_file]`.

Reply:
I must have made that up.

Reply:
Added an `inventory_source` method to the CLI.